### PR TITLE
Create skeleton for sitewide notices block.

### PR DESCRIPTION
**Pull request by dcavins**
Here is the core code for creating and outputting the sitewide notices block. @imath if you have a good block example of adding the functional JS to do a little task (in this case dismiss the notice), please let me know what that is and I'll add the JS dismiss action to this block. Thanks!

### CODE REVIEW

File: build/bp-messages/bp-messages-blocks.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+		return;
+	}
+
+	$closed_notices = bp_get_user_meta( bp_loggedin_user_id(), 'closed_notices', true );
+
+	if ( empty( $closed_notices ) ) {
+		$closed_notices = array();
+	}
+
+	if ( is_array( $closed_notices ) && in_array( $notice->id, $closed_notices )  ) {
```
Comment (renatonascalves) [inline build/bp-messages/bp-messages-blocks.php:87]: Maybe do `$closed_notices = (array) bp_get_user_meta( bp_loggedin_user_id(), 'closed_notices', true );`?

Removing the conditional.

File: build/bp-messages/bp-messages-blocks.php [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Register the friends blocks.
+ *
+ * @since 9.0.0
+ */
+function register_messages_blocks() {
+	$blocks = array(
+		'bp/friends' => array(
```
Comment (renatonascalves) [inline build/bp-messages/bp-messages-blocks.php:23]: Is `bp/friends` correct here?

**Review by renatonascalves — CHANGES_REQUESTED** [review]
Some questions and suggestions!

File: build/bp-messages/bp-messages-blocks.php (hunk 2 above)
Comment (dcavins) [reply build/bp-messages/bp-messages-blocks.php]: Not at all! :)

File: build/bp-messages/bp-messages-blocks.php [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+/**
+ * Register the messages component blocks.
+ *
+ * @since 9.0.0
+ */
+function register_messages_blocks() {
+	$blocks = array(
+		'bp/messages' => array(
```
Comment (imath) [inline build/bp-messages/bp-messages-blocks.php:23]: It should be the name of the block here 😄 
```suggestion
		'bp/sitewide-notices' => array(
```

File: build/bp-messages/bp-messages-blocks.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+				<strong>%2$s</strong>
+				<button type="button" id="close-notice" class="bp-tooltip" data-bp-tooltip="%3$s" data-bp-sitewide-notice-id="%4$d"><span class="bp-screen-reader-text">%5$s</span> <span aria-hidden="true">&Chi;</span></button>
+				%6$s
+			</div>
+		</div>',
+		esc_attr( $notice->id ),
+		bp_get_message_notice_subject( $notice ),
+		esc_attr__( 'Dismiss this notice', 'buddypress' ),
+		esc_attr( $notice->id ),
+		__( 'Dismiss this notice', 'buddypress' ),
```
Comment (imath) [inline build/bp-messages/bp-messages-blocks.php:118]:
```suggestion
		esc_html__( 'Dismiss this notice', 'buddypress' ),
```

File: build/bp-messages/bp-messages-blocks.php (hunk 1 above)
Comment (imath) [reply build/bp-messages/bp-messages-blocks.php:87]: Yeah +1

File: src/bp-messages/js/blocks/sitewide-notices.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+		background: '#fff',
+		foreground: '#d84800',
+		src: 'megaphone',
+	},
+	category: 'buddypress',
+	attributes: {
+	title: {
+		type: 'string',
+		default: '',
+		},
```
[the remark this answers is not in the document]
Comment (imath) [reply src/bp-messages/js/blocks/sitewide-notices.js:32]: Are you sure? It looks fine imho.

**Review by imath — COMMENTED** [review]
Hi @dcavins 

Thanks a lot for your contributions 💪. It looks nice, I've added some suggestions. I'll try to test it asap.

File: src/bp-messages/js/blocks/sitewide-notices.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (dcavins) [reply src/bp-messages/js/blocks/sitewide-notices.js:32]: There was an alignment problem that was obvious if you could see the surrounding code. I fixed it.

File: build/bp-messages/bp-messages-blocks.php (hunk 1 above)
Comment (dcavins) [reply build/bp-messages/bp-messages-blocks.php:87]: Sure! Maybe we should change the code in BP where I got that logic from, too.
`bp_message_get_notices()` and `bp_nouveau_push_sitewide_notices()` are similar.

File: src/bp-messages/js/blocks/sitewide-notices.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (imath) [reply src/bp-messages/js/blocks/sitewide-notices.js:32]: Great :)

**Review by imath — COMMENTED** [review]
This looks very promising !! Thanks a lot for your great work on this @dcavins 💪. I'll test it asap.

**Pull request merged**